app/_gateway_entities/sni.md

```diff
@@ -31,6 +31,29 @@ schema:
 
 An SNI (Server Name Indication) is used to map multiple hostnames to a [Certificate](/gateway/entities/certificate/). It allows {{site.base_gateway}} to select which SSL/TLS Certificate to use based on the hostname in the client request. This feature ensures that multiple domains can be securely served through the same gateway.
 
+## SNI routing
+
+When configuring a route with a secure protocol like HTTPS, gRPC, or TLS, an SNI can be used for routing. The SNI is determined during the TLS handshake process and will remain unchanged for the duration of the connection, so all requests will contain the same SNI regardless of the defined `Header` in the Route configuration. For more information on how routing priorities are assigned read the [Expressions Router](/gateway/routing/expressions/#performance-considerations) documentation.
+
+### Wildcards 
+
+Valid wildcard positions for configuring SNIs are: 
+
+* `mydomain.*`
+* `*.mydomain.com`
+* `*.www.mydomain.com`
+
+This is especially useful when configuring [TLS Routes](/gateway/entities/#tls-route-configuration). 
+### Prioritization matching
+
+The prioritization for matching `snis` follows this order:
```

Comment:
```suggestion
The prioritization for matching SNIs follows this order:
```
Clarify _what_ is matching the SNIs. Based on how this is written, I think this is matching the SNI in a Route to Certificates.